projects/roscoe/roscoe_data/restore_annotated.py

```diff
@@ -0,0 +1,391 @@
+                struct["gpt-3"] = reasoning
+                # TODO: TMP, remove!
+                struct["dataset"] = "cosmos_valid_gpt3_expl.NEW"
+                structs.append(struct)
+            else:
+                val = struct["key"]
+                print(f"No reasoning found for key {val}")
+
+            if len(structs) > 200:
+                break
+
+    write_to_file(structs, savefile)
+
+
+def parse_semeval(fn, reasoning, savefile):
+    json_lines = []
+    reasonings = {}
+    with open(reasoning, "r") as f:
+        json_lines = json.load(f)
+
+    for json_line in json_lines:
+        reasonings[json_line["key"]] = json_line
+
+    with open(fn) as fd:
+        doc = xmltodict.parse(fd.read())
+
+    structs = []
+    data = doc['data']  # == u'an attribute'
+    instances = data['instance']  # == [u'elements', u'more elements']
+    cnt = 0
+    for instance in instances:
+        ins = {}
+        context = re.sub(r"\s+", ' ', instance["text"])
+        id = instance["@id"]
+
+        ins["id"] = id
+        ins["premise"] = context
+
+        ins["questions"] = []
+
+        for question in instance['questions']["question"]:
+            quu1 = {
+                "premise": "",
+                "hypothesis": "",
+                "answer": "",
+                "gpt-3": "",
+                "dataset": "",
+            }
+            quu2 = {
+                "premise": "",
+                "hypothesis": "",
+                "answer": "",
+                "gpt-3": "",
+                "dataset": "",
+            }
+
+            q = question["@text"]
+            quu1["key"] = (
+                instance["@id"]
+                + "\t"
+                + question["@id"]
+                + "\t"
+                + question["answer"][0]["@id"]
+            )
+            quu2["key"] = (
+                instance["@id"]
+                + "\t"
+                + question["@id"]
+                + "\t"
+                + question["answer"][1]["@id"]
+            )
+
+            if question["answer"][0]["@correct"] == "False":
+                quu1["hypothesis"] = question["answer"][0]["@text"]
+                quu1["answer"] = "no"
+            else:
+                quu1["hypothesis"] = question["answer"][0]["@text"]
+                quu1["answer"] = "yes"
+
+            if question["answer"][1]["@correct"] == "False":
+                quu2["answer"] = "no"
+                quu2["hypothesis"] = question["answer"][1]["@text"]
+            else:
+                quu2["hypothesis"] = question["answer"][1]["@text"]
+                quu2["answer"] = "yes"
+
+            quu1["premise"] = context + " " + q
+            quu2["premise"] = context + " " + q
+
+            if quu1["key"] in reasonings:
+                reasoning = reasonings[quu1["key"]]["reasoning"]
+                quu1["gpt-3"] = reasoning
+                # TODO: TMP, remove!
+                quu1["dataset"] = "semevalcommonsense_gpt3_expl"
```

Comment:
This the right path?

Reply:
lol, thank you for looking! this is not the path, that's key-value in the output JSON I added to completely match with your files, so I could easily run "diff". Will remove!